php/connect/class-api.php

```diff
@@ -409,6 +409,8 @@ function_exists( 'wp_get_original_image_url' ) &&
 		// Check if we can try http file upload.
 		if ( ( empty( $headers ) && empty( $disable_https_fetch ) && true === $try_remote ) || ! $media->is_uploadable_media( $attachment_id ) ) {
 			$args['file'] = $file_url;
+		} elseif ( ! $media->is_local_media( $attachment_id ) ) {
```

Comment:
@DavidCramer 

I'm still not quite happy here. WDYT if we introduce yet another method called something like `is_external_media`, and the `is_uploadable_media` would be a combination of that and `is_local_media`. This way we can simplify the initial condition to something like `if ( ( empty( $headers ) && empty( $disable_https_fetch ) && true === $try_remote ) || $media->is_external_media( $attachment_id ) ) {`. I suspect that this would be helpful in the future.

Reply:
@pereirinha - I'm actually thinking of splitting the upload method completely to be like, `upload_local` and `upload_remote` then determine which to use at the `upload_asset` level.
I'll do a PR to prototype it.